assets/filter_repos.js

```diff
@@ -0,0 +1,37 @@
+const searchInput = document.getElementById('search-hook-id');
+const selectInput = document.getElementById('select-hook-type');
+
+const hooks = document.getElementById('hooks');
+const repos = hooks.getElementsByTagName('ul');
+
+const filterHooks = () => {
+    const id = searchInput.value.toLowerCase();
+    const type = selectInput.value;
+
+    for (let i = 0; i < repos.length; i += 1) {
+        const repo = repos[i];
+        let hasVisibleHooks = false;
+        const repoHooks = repo.getElementsByTagName('li');
+
+        if (repoHooks) {
+            for (let j = 0; j < repoHooks.length; j += 1) {
+                const repoHook = repoHooks[j];
+                const hookId = repoHook.dataset.id.toLowerCase();
+                const hookTypes = repoHook.dataset.types;
+
+                if (hookId.includes(id) && hookTypes.includes(type)) {
+                    repoHook.hidden = false;
+                    hasVisibleHooks = true;
+                } else {
+                    repoHook.hidden = true;
+                }
+            }
+        }
+
+        repo.hidden = !hasVisibleHooks;
+        hooks.querySelector(`h3[data-repo="${repo.dataset.repo}"]`).hidden = !hasVisibleHooks;
+    }
+};
+
+searchInput.addEventListener('input', filterHooks);
+selectInput.addEventListener('change', filterHooks);
```

Comment:
will need to look at caniuse for:
- `let`
- fat arrow
- format strings
- `const`
- `.includes(...)`
- `.dataset`

I suspect at least one of those things will need to be made more-compatible

Reply:
the whole code fits the ES6, detailed caniuse:

* `let`: [96.75%](https://caniuse.com/let)
* fat arrow: [95.47%](https://caniuse.com/arrow-functions)
* format strings: [95.99%](https://caniuse.com/template-literals)
* `const`: [99.16%](https://caniuse.com/const)
* `.includes(...)`: [95.98%](https://caniuse.com/es6-string-includes)
* `.dataset`: [99.54%](https://caniuse.com/dataset)

to meet `>0.2%, not dead`, `const` and `let` have to be changed to var and the template literal with e.g. `.concat`.